common/util.c

```diff
@@ -64,3 +68,36 @@ int get_modifier_names(const char **names, uint32_t modifier_masks) {
 
 	return length;
 }
+
+pid_t get_parent_pid(pid_t child) {
+	pid_t parent;
+	char file_name[100];
+	char *buffer = NULL;
+	char *token = NULL;
+	const char sep[2] = " ";
+	FILE *stat = NULL;
+
+	sway_log(L_DEBUG, "trying to get parent pid for child pid %d", child);
+
+	sprintf(file_name, "/proc/%d/stat", child);
+
+	if (!(stat = fopen(file_name, "r")) || !(buffer = read_line(stat))) {
```

Comment:
I guess either would work in this case, since the `read_line` will always fail if the stat file can't be opened. I'll change it, that will prevent even checking the second condition.